modules/renter/hostdb/scan.go

```diff
@@ -78,6 +78,56 @@ func (hdb *HostDB) decrementReliability(addr modules.NetAddress, penalty types.C
 	}
 }
 
+// managedUpdateEntry updates an entry in the hostdb after a scan has taken
+// place.
+func (hdb *HostDB) managedUpdateEntry(hostEntry *hostEntry, settings modules.HostExternalSettings, err error) {
+	hdb.mu.Lock()
+	defer hdb.mu.Unlock()
+
+	// Regardless of whether the host responded, add it to allHosts.
+	priorHost, exists := hdb.allHosts[hostEntry.NetAddress]
+	if !exists {
+		hdb.allHosts[hostEntry.NetAddress] = hostEntry
```

Comment:
I think the conditional "hdb.allHosts[hostEntry.NetAddress] = hostEntry" assignment here may be causing a problem. (Not that this is related to the issue we're patching in this ticket - and this is how it pas previously, not changed by this patch).

I think this can leave the AllHosts out of date with respect to the ActiveHosts. The idea is that when AllHosts/ActiveHosts have been loaded from the persistent store, the hostEntry pointer in allHosts and the one in the node within activeHosts (for the same network address) point to separate (but similar) structures. The conditional assigment here means that subsequent updates in settings discovered during the scan may not be set in the hostEntry accessible via allHosts. Maybe the assignment should be done after the "if err != null {" bock, and done unconditionally (i.e. done even if allHosts already has the network address).

This may be involved in ticket #1320; if the renter uses an out of date payment or collateral rate setting when making a contract revision or renewal the host may reject it due to incorrect payment. I've seen some requests at my host that might be consistent with that. (The values sent in the revision or renewal corresponded to a price or collateral rate setting I had since changed).

I'll open a new ticket if I can write a test to show the problem..